### PR TITLE
Add update sieve responsible for package updates

### PR DESCRIPTION
**Pull request by fridex**
## Related Issues and Dependencies

Fixes: https://github.com/thoth-station/adviser/issues/1886

## This introduces a breaking change

- [x] No

## Description

With this pipeline unit, we will be able to perform updates of application stacks as triggered by the internal trigger.

### CODE REVIEW

**Comment by KPostOffice** [comment]
This is a great idea. One question, if the update type isn't latest and the new package causes the stack to score worse than the original will this still advise the original stack?

**Comment by fridex** [comment]
> This is a great idea. One question, if the update type isn't latest and the new package causes the stack to score worse than the original will this still advise the original stack?

I've added a warning to the implementation in b23e549. In some cases, it might still be reasonable to resolve "performance" or "stable" software and consider only the newly issued release. E.g. if I'm running `tensorflow==2.4.0` on performance recommendation type and `tensorflow==2.5.0` is released, adviser can still look for a performant stack given the `tensorflow==2.5.0` release.

**Comment by fridex** [comment]
> So the idea here is when we have an internal trigger for a package release we will propagate it through this environment variable?

Yes, exactly. Package releases spots a new release -> sends a message with (`package_name`, `package_version`, `index_url`) -> investigator receives the given message, queries the database for stacks that use `package_name` in stacks and schedules adviser with (`package_name`, `package_version`, `index_url`) -> adviser computes results -> as the advise workflow was scheduled based on a new release -> kebechet opens a pull request if the stack needs to be updated.

Besides this workflow, [user's software stack is considered as a base for scoring](https://thoth-station.ninja/j/user_stack.html).

File: tests/sieves/test_update.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+    UNIT_TESTED = PackageUpdateSieve
+    _PACKAGE_UPDATE_DICT = {
+        "package_name": "tensorflow",
+        "package_version": "2.5.0",
+        "index_url": "https://pypi.org/simple",
+    }
+
+    @pytest.mark.skip(
+        reason="Skip default configuration as default is not considered valid and needs futher adjustments"
```
Comment (KPostOffice) [inline tests/sieves/test_update.py:45]:
```suggestion
        reason="Skip default configuration as default is not considered valid and needs further adjustments"
```

**Review by KPostOffice — COMMENTED** [review]
Just noticed a small spelling mistake otherwise I think this is good 2 go.

**Review by KPostOffice — APPROVED** [review]

**Pull request merged**